mono/metadata/appdomain.c

```diff
@@ -2444,6 +2470,18 @@ mono_domain_assembly_preload (MonoAssemblyLoadContext *alc,
 	req.request.predicate = predicate;
 	req.request.predicate_ud = predicate_ud;
 
+#ifdef ENABLE_NETCORE
+	char **search_path = g_new0 (char *, 2);
+	search_path [1] = NULL;
+
+	char *base_dir = get_app_context_base_directory (error);
```

Comment:
Probably need a `mono_runtime_get_no_exec()` early exit for AOT somewhere around here.